logstash-core/lib/logstash/runner.rb

```diff
@@ -184,6 +198,21 @@ def configure_plugin_paths(paths)
     end
   end
 
+  def agent_class
+    return @agent_class if @agent_class # We don't want to load these things twice
```

Comment:
a lookup should be rather harmless so I suggest:

* eliminating this method
* make create agent:

```ruby
def create_agent
  @logger.info("Loading agent", :class => agent_class)
  LogStash::AgentPluginRegistry.lookup(agent_name).new(@logger)
end
```

* make the logic of raising a `MissingAgentError` exception part of the lookup method
* put the @logger.fatal message in the rescue of `@agent = create_agent`